tests/unit/test_filter.py

```diff
@@ -292,3 +306,179 @@ def test_num_filter_zero():
+
+
+def test_timestamp_operators():
+    """Test all comparison operators for Timestamp filter."""
+    dt = datetime(2023, 3, 17, 14, 30, 0, tzinfo=timezone.utc)
+    ts_value = dt.timestamp()
+
+    # Equal
+    ts = Timestamp("created_at") == dt
+    assert str(ts) == f"@created_at:[{ts_value} {ts_value}]"
+
+    # Not equal
+    ts = Timestamp("created_at") != dt
+    assert str(ts) == f"(-@created_at:[{ts_value} {ts_value}])"
+
+    # Greater than
+    ts = Timestamp("created_at") > dt
+    assert str(ts) == f"@created_at:[({ts_value} +inf]"
+
+    # Less than
+    ts = Timestamp("created_at") < dt
+    assert str(ts) == f"@created_at:[-inf ({ts_value}]"
+
+    # Greater than or equal
+    ts = Timestamp("created_at") >= dt
+    assert str(ts) == f"@created_at:[{ts_value} +inf]"
+
+    # Less than or equal
+    ts = Timestamp("created_at") <= dt
+    assert str(ts) == f"@created_at:[-inf {ts_value}]"
+
+    td = timedelta(days=5)
+    dt2 = dt + td
+    ts_value2 = dt2.timestamp()
+
+    ts = Timestamp("created_at").between(dt, dt2)
+    assert str(ts) == f"@created_at:[{ts_value} {ts_value2}]"
+
+    ts = Timestamp("created_at").between(dt, dt2, inclusive="neither")
+    assert str(ts) == f"@created_at:[({ts_value} ({ts_value2}]"
+
+    ts = Timestamp("created_at").between(dt, dt2, inclusive="left")
+    assert str(ts) == f"@created_at:[{ts_value} ({ts_value2}]"
+
+    ts = Timestamp("created_at").between(dt, dt2, inclusive="right")
+    assert str(ts) == f"@created_at:[({ts_value} {ts_value2}]"
+
+
+def test_timestamp_between():
+    """Test the between method for date ranges."""
+    start = datetime(2023, 3, 1, 0, 0, 0, tzinfo=timezone.utc)
+    end = datetime(2023, 3, 31, 23, 59, 59, tzinfo=timezone.utc)
+
+    ts = Timestamp("created_at").between(start, end)
+
+    start_ts = start.timestamp()
+    end_ts = end.timestamp()
+
+    assert str(ts) == f"@created_at:[{start_ts} {end_ts}]"
+
+    # Test with dates (should expand to full days)
+    start_date = date(2023, 3, 1)
+    end_date = date(2023, 3, 31)
+
+    ts = Timestamp("created_at").between(start_date, end_date)
+
+    # Start should be beginning of day
+    expected_start = datetime.combine(start_date, datetime.min.time())
+    expected_start = expected_start.replace(tzinfo=timezone.utc)
+
+    # End should be end of day
+    expected_end = datetime.combine(end_date, datetime.max.time())
+    expected_end = expected_end.replace(tzinfo=timezone.utc)
+
+    expected_start_ts = expected_start.timestamp()
+    expected_end_ts = expected_end.timestamp()
+
+    assert str(ts) == f"@created_at:[{expected_start_ts} {expected_end_ts}]"
+
+
+def test_timestamp_none():
+    """Test handling of None values."""
+    ts = Timestamp("created_at") == None
+    assert str(ts) == "*"
+
+    ts = Timestamp("created_at") != None
+    assert str(ts) == "*"
+
+    ts = Timestamp("created_at") > None
+    assert str(ts) == "*"
+
+
+def test_timestamp_invalid_input():
```

Comment:
On a similar theme, what happens when the `Timestamp` field is called / created for a field that isn't numeric in the index? How does that behave? Given there is a requirement that `Timestamp` filters are tied to `num` fields, thinking through how we should document that too.